util/reciprocal_lattice/__init__.py

```diff
@@ -87,6 +90,16 @@ def load_models(self, experiments, reflections):
                 for c in crystals
             ]
             self.viewer.set_reciprocal_lattice_vectors(vecs)
+            vecs = [
+                [
+                    matrix.col(l)
+                    for l in (100 * matrix.sqr(c.get_B()))
+                    .transpose()
+                    .as_list_of_lists()
+                ]
+                for c in crystals
+            ]
+            self.viewer.set_reciprocal_crystal_vectors(vecs)
```

Comment:
The similar block immediately above, which defines `vecs` the first time, should be removed after this is added

Reply:
Not true, I believe 

```
        if crystals:
            # the points are scaled by 100 so must do that here too
            vecs = [
                [
                    matrix.col(l)
                    for l in (100 * matrix.sqr(c.get_A()))
                    .transpose()
                    .as_list_of_lists()
                ]
                for c in crystals
            ]
            self.viewer.set_reciprocal_lattice_vectors(vecs)
            vecs = [
                [
                    matrix.col(l)
                    for l in (100 * matrix.sqr(c.get_B()))
                    .transpose()
                    .as_list_of_lists()
                ]
                for c in crystals
            ]
            self.viewer.set_reciprocal_crystal_vectors(vecs)
```

the first transforms the `A` matrices, sets them as one property, the second does the same for `B` - I _could_ refactor the code significantly to carry the U matrices to where the cell drawing is done, but this would be a major overhaul.

Reply:
Right, I see. At a quick glance I did not see that a different method was called after the listcomp :roll_eyes: